### ISSUE TITLE
unneeded_synthesized_initializer rule does not detect violations in a struct nested inside a class

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#5120]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given

```
class Foo {
    struct Bar {
        let baz: Int

        init(baz: Int) {
            self.baz = baz
        }
    }
}
```

`unneeded_synthesized_initializer` will fail to detect the violation here.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.52.4
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods

* Which Xcode version are you using (check `xcodebuild -version`)?

14.2

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

See above
